client/src/recoil/story/selector.ts

```diff
@@ -2,29 +2,43 @@ import { DefaultValue, selectorFamily } from 'recoil';
 import produce from 'immer';
 import storyListAtom from './atom';
 import { StoryType } from '@/types/story';
+import { getTasksByStoryId } from '@/lib/api/task';
 
 export const storyState = selectorFamily<StoryType | undefined, number>({
   key: 'storyWithIdSelector',
-  get: (id) => ({ get }) => {
-    const storyList = get(storyListAtom);
-    return storyList?.find((story) => story.id === id);
-  },
-  set: (id: number) => ({ set, get }, newValue) => {
-    set(
-      storyListAtom,
-      newValue instanceof DefaultValue
-        ? get(storyListAtom)
-        : (prev) =>
-            produce(prev, (draft) => {
-              const newStory = draft?.find((story) => story.id === id);
-              if (!newStory || !newValue) return;
-              newStory.name = newValue.name;
-              newStory.id = newValue.id;
-              newStory.order = newValue.order;
-              newStory.status = newValue.status;
-              newStory.epicId = newValue.epicId;
-              newStory.projectId = newValue.projectId;
-            }),
-    );
+  get:
+    (id) =>
+    ({ get }) => {
+      const storyList = get(storyListAtom);
+      return storyList?.find((story) => story.id === id);
+    },
+  set:
+    (id: number) =>
+    ({ set, get }, newValue) => {
+      set(
+        storyListAtom,
+        newValue instanceof DefaultValue
+          ? get(storyListAtom)
+          : (prev) =>
+              produce(prev, (draft) => {
+                const newStory = draft?.find((story) => story.id === id);
+                if (!newStory || !newValue) return;
+                newStory.name = newValue.name;
+                newStory.id = newValue.id;
+                newStory.order = newValue.order;
+                newStory.status = newValue.status;
+                newStory.epicId = newValue.epicId;
+                newStory.projectId = newValue.projectId;
```

Comment:
원래 코드는 해수님이 작성하신게 아닌것 같지만 얘는 `findIndex`로 간단하게 표현할 수 있을것같아요

그와중에 `findIndex`는 탐색 실패시 -1를 반환하는게 너무 거슬리네요ㅋㅋㅋ

```suggestion
                const newStoryIdx = draft?.findIndex((story) => story.id === id);
                if (newStoryIdx === -1 || !newValue) return;
                draft[newStoryIdx] = newValue;
```

Reply:
수정했습니다 !